### PR TITLE
chore: Add check job to CI workflow to ensure all tests have passed

### PR DESCRIPTION
**Pull request by mattkram**
This allows a single stage to be used for branch protection to ensure all tests pass before a PR can be merged. Otherwise, we would need to specify each entry in the test matrix manually, which is tedious and error-prone.

### CODE REVIEW

**Comment by AlbertDeFusco** [comment]
Fine by me. Let me know when you're ready for review and merge.

**Comment by mattkram** [comment]
I'm just planning to add a branch rule and confirm this works for that, then I'll merge. I can do that myself, or if you would like to give me a ✔️ that'd be good too 🙂

**Review by AlbertDeFusco — APPROVED** [review]

**Comment by AlbertDeFusco** [comment]
Approved, go ahead and merge when you are ready

**Comment by mattkram** [comment]
Okay. Branch protection is enabled such that the `check` job must pass before a PR can be merged.

Admins can override this if necessary.

**Pull request merged**
